### PR TITLE
Feature/components laura

### PR DESCRIPTION
**Pull request by lalonggone**
# Pull Request 

## Description of Changes

This PR organizes most the app's routes and component architecture. Big push on styling as well. There is some conditional rendering logic on the HomePage and NavBar components worth checking out (it's working, and looks clean to me, but def had to take a minute to digest it)

## Type of Change

- [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
- [x] 🔮 New feature (non-breaking change which adds functionality)
- [ ] 📖 Documentation Update
- [ ] Other (please describe):

## Has this been Tested? 🧪

We don't have official tests written yet, that is coming soon to an iteration near you! But we did walk through the code together with the app open on the browser  and everything seems to be working as expected so far.

## Checklist ✅

- [x] I have performed a self-review of my own code.
- [x] I have made notes of my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [na] I have added tests that prove the new feature works if necessary.
- [na] All tests are passing.

## Screenshots/Recordings 📸
Screenshot 2024-04-05 at 5 07 00 PM

## TO DO and Additional Comments 💬

- finish building the rest of components
- error handling
- Cypress testing
- prop-types
- connecting the API
- fine tune the actual data set in the API

### CODE REVIEW

File: src/components/HomePage/HomePage.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
+import React from 'react'
+import { useState } from 'react'
+import { useNavigate } from 'react-router-dom'
```
Comment (dkwon1223) [inline src/components/HomePage/HomePage.js:3]: Love this new hook!

File: src/components/NavBar/NavBar.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import React, { useState } from 'react'
+import { Link, useLocation } from 'react-router-dom'
+import './NavBar.css'
+
+function NavBar() {
+  const location = useLocation()
+
+  const initialShowAllDrinks = location.pathname === '/all'
```
Comment (dkwon1223) [inline src/components/NavBar/NavBar.js:8]: Interesting use of state!

File: src/components/HomePage/HomePage.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+import './HomePage.css'
+
+function HomePage() {
+  const [drinkType, setDrinkType] = useState('')
+  const navigate = useNavigate()
```
Comment (dkwon1223) [inline src/components/HomePage/HomePage.js:8]: Cool new hook!

File: src/components/NavBar/NavBar.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  const [showAllDrinks, setShowAllDrinks] = useState(initialShowAllDrinks)
```
Comment (dkwon1223) [inline src/components/NavBar/NavBar.js:9]: Love this use of state

File: src/components/AllDrinksGrid/AllDrinksGrid.css
Comment (crotteau) [inline src/components/AllDrinksGrid/AllDrinksGrid.css:1]: love the background styling you've added

File: src/components/AllDrinksGrid/AllDrinksGrid.js [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+      </section>
+        <NavBar />
+      <section className="drinks-grid">
+        {mixedDrinkRecipes.map((recipe) => {
+          return (
+            <div className="drink-card" key={recipe.id}>
+              <img src={recipe.image} alt={recipe.name} />
+              <h2>{recipe.name}</h2>
+              {/* <ul>
+                {recipe.ingredients.map((ingredient, index) => {
```
Comment (crotteau) [inline src/components/AllDrinksGrid/AllDrinksGrid.js:21]: I'll add a filter function for the alcoholic drinks here

File: src/components/HomePage/HomePage.js [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+      </section>
+      <section className="quiz-section">
+        <h1 className="quiz-title">Do you prefer...</h1>
+        <div className="radio-container">
+          <label className="radio">
+              Alcoholic beverages
+            <input
+              type="radio"
+              name="drinkType"
+              value="alcoholic"
```
Comment (crotteau) [inline src/components/HomePage/HomePage.js:32]: I think I'll change the values to a boolean "true" or "false" so the state is reusable for the filter function

File: src/components/HomePage/HomePage.js (hunk 1 above)
Comment (lalonggone) [reply src/components/HomePage/HomePage.js:3]: right?! I was like... it can't be that easy.. but it is! 😆

**Review by crotteau — APPROVED** [review]

File: src/components/HomePage/HomePage.js (hunk 6 above)
Comment (lalonggone) [reply src/components/HomePage/HomePage.js:32]: that sounds like a good plan. you might have to update the handleClick function logic? 
excited to see what you come up with!

File: src/components/AllDrinksGrid/AllDrinksGrid.js [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+function AllDrinksGrid() {
+
+ return (
+    <div>
+      <section className="home-title">
+        <h1>Mix Master</h1>
+      </section>
+        <NavBar />
+      <section className="drinks-grid">
+        {mixedDrinkRecipes.map((recipe) => {
```
Comment (kaylagordon) [inline src/components/AllDrinksGrid/AllDrinksGrid.js:15]: Might be worth destructuring `recipe` here so you don't have to keep referencing `recipe.` in the next several lines of code.

File: src/components/App/App.js [hunk 8]
```diff
@@ -1,9 +1,27 @@
...
-
+      <BrowserRouter>
+      <Routes>
+        <Route path="/" element={<HomePage />} />
+        <Route path="/all" element={<AllDrinksGrid />} />
+        <Route path="/na" element={<NaDrinksGrid />} />
+        <Route path="/favorites" element={<FavoriteDrinksGrid />} />
+        {/* <Route path="/random" element={<RandomDrink />} /> */}
+        {/* <Route path="/drink/:id" element={<DrinkDetails />} /> */}
+        {/* <Route path="*" element={<NotFound />} /> */}
```
Comment (kaylagordon) [inline src/components/App/App.js:22]: I'm assuming these paths will be added back in at some point?

File: src/components/FavoriteDrinksGrid/FavoriteDrinksGrid.js [hunk 9]
```diff
@@ -0,0 +1,17 @@
+import React from 'react'
+import DrinkDetails from '../DrinkDetails/DrinkDetails'
+import NavBar from '../NavBar/NavBar'
+
+function RecipeGrid() {
+  return (
+    <div>
```
Comment (kaylagordon) [inline src/components/FavoriteDrinksGrid/FavoriteDrinksGrid.js:7]: I'm noticing a lot of `divs` as your outer-most wrappers in your JSX. Have you looked into using [fragments](https://react.dev/reference/react/Fragment)? Or are there more semantic HTML elements to reach for?

File: src/components/HomePage/HomePage.js [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+import { useNavigate } from 'react-router-dom'
+import './HomePage.css'
+
+function HomePage() {
+  const [drinkType, setDrinkType] = useState('')
+  const navigate = useNavigate()
+
+  const handleChoice = (choice) => {
+    if (choice === 'alcoholic') {
+      navigate('/all')
```
Comment (kaylagordon) [inline src/components/HomePage/HomePage.js:12]: According to the [router docs](https://reactrouter.com/en/main/hooks/use-navigate), it's better to use `redirect` in cases like this.

File: src/components/NaDrinksGrid/NaDrinksGrid.js [hunk 11]
```diff
@@ -0,0 +1,16 @@
+import React from 'react'
+import NavBar from '../NavBar/NavBar'
+// import DrinkDetails from '../DrinkDetails/DrinkDetails'
+
+function NaDrinksGrid() {
+  return (
+    <div>
+       <section className="home-title">
+        <h1>Mix Master</h1>
+      </section>
```
Comment (kaylagordon) [inline src/components/NaDrinksGrid/NaDrinksGrid.js:10]: indentation is off here

File: src/components/NavBar/NavBar.js [hunk 12]
```diff
@@ -0,0 +1,38 @@
...
+            <Link to={showAllDrinks ? '/na' : '/all'}>
+              {showAllDrinks ? 'NA Drinks' : 'All Drinks'}
+            </Link>
+          </li>
+          <li className="nav-item">
+            <Link to="/favorites">Favorite Drinks</Link>
+          </li>
+        </ul>
+      </nav>
+        <div className="border"></div>
```
Comment (kaylagordon) [inline src/components/NavBar/NavBar.js:33]: indentation is off. It also looks like you have a `div` wrapped around another `div`? Is there a reason you have two of them there?

**Pull request merged**